pennylane/debugging/operation_checker.py

```diff
@@ -0,0 +1,833 @@
+
+    Returns:
+        callable: The wrapped method of the operation.
+
+    The object returned by the returned callable differs, depending on the scenario:
+
+      - If the method succeeds, the return value of the method is returned,
+      - If the ``expected_exc`` is raised, ``None`` is returned,
+      - If another exception is raised, it is returned (but not raised).
+    """
+    _method = getattr(op, method)
+
+    def wrapped_method(*args, **kwargs):
+        r"""Wrapped operation method that tolerates an expected exception
+        and catches (and returns) all other exceptions."""
+        try:
+            return _method(*args, **kwargs)
+        except expected_exc:
+            return None
+        except Exception as e:
+            return e
+
+    return wrapped_method
+
+
+def matrix_from_matrix(op, par, wires):
+    r"""Get the matrix of an operation, using ``get_matrix``.
+
+    Args:
+        op (type): Operation type to obtain the matrix for
+        par (array_like): Parameters of the operation
+        wires (.wires.Wires): Wires of the operation
+
+    Returns:
+        object: Matrix of the operation if it is defined and no problem occured
+        with ``get_matrix``. ``None`` if no matrix is defined via ``get_matrix``
+        or ``Exception`` if an error occured.
+    """
+    instance = op(*par, wires=wires)
+    return wrap_op_method(instance, "get_matrix", MatrixUndefinedError)()
+
+
+def matrix_from_sparse_matrix(op, par, wires):
+    r"""Get the matrix of an operation, using ``sparse_matrix``.
+
+    Args:
+        op (type): Operation type to obtain the matrix for
+        par (array_like): Parameters of the operation
+        wires (.wires.Wires): Wires of the operation
+
+    Returns:
+        object: Matrix of the operation if it is defined and no problem occured
+        with ``sparse_matrix``. ``None`` if no matrix is defined via ``sparse_matrix``
+        or ``Exception`` if an error occured.
+    """
+    instance = op(*par, wires=wires)
+    smat = wrap_op_method(instance, "sparse_matrix", SparseMatrixUndefinedError)()
+    if smat is None or isinstance(smat, Exception):
+        return smat
+    return qml.matrix(smat)
+
+
+def matrix_from_terms(op, par, wires):
+    r"""Get the matrix of an operation, using its ``terms``.
+
+    Args:
+        op (type): Operation type to obtain the matrix for
+        par (array_like): Parameters of the operation
+        wires (.wires.Wires): Wires of the operation
+
+    Returns:
+        object: Matrix of the operation if it is defined and no problem occured
+        with ``terms``. ``None`` if no terms are defined via ``terms``
+        or ``Exception`` if an error occured.
+    """
+    instance = op(*par, wires=wires)
+    terms = wrap_op_method(instance, "terms", TermsUndefinedError)()
+    if terms is None or isinstance(terms, Exception):
+        return terms
+
+    return np.sum([c * qml.matrix(op) for c, op in zip(*terms)], axis=0)
```

Comment:
Alternatively, you could use the terms to create a Hamiltonian, and just call `qml.matrix(H)`. This way the utility function does not include its own logic.